### PR TITLE
Implement Scenario::reset

### PR DESCRIPTION
**Pull request by ZehMatt**

### CODE REVIEW

File: src/OpenLoco/CompanyManager.cpp [hunk 1]
```diff
@@ -26,6 +26,11 @@ namespace OpenLoco::CompanyManager
 
     static void produceCompanies();
 
+    void reset()
+    {
+        call(0x0042F7F8);
```
Comment (duncanspumpkin) [inline src/OpenLoco/CompanyManager.cpp:31]: for all of your functions add the address as a comment above the function.

**Review by duncanspumpkin — APPROVED** [review]
Could you add the comments and thats about all thats needed

File: src/OpenLoco/Title.cpp [hunk 2]
```diff
@@ -1,4 +1,4 @@
-#include "Title.h"
+﻿#include "Title.h"
```
Comment (AaronVanGeffen) [inline src/OpenLoco/Title.cpp:1]: Byte-order mark insertion?

File: src/OpenLoco/Title.h [hunk 3]
```diff
@@ -1,7 +1,8 @@
-#pragma once
+﻿#pragma once
```
Comment (AaronVanGeffen) [inline src/OpenLoco/Title.h:1]: Byte-order mark insertion?

File: src/OpenLoco/Scenario.cpp [hunk 4]
```diff
@@ -24,10 +31,84 @@ namespace OpenLoco::Scenario
...
+
+        CompanyManager::reset();
+        StringManager::reset();
+        ThingManager::reset();
+
+        Ui::Windows::Construction::Construction::reset();
+        sub_46115C();
+        sub_4C4BC0();
+
+        sub_49685C(0x76C);
```
Comment (AaronVanGeffen) [inline src/OpenLoco/Scenario.cpp:99]: This is changing the year.
```suggestion
        sub_49685C(1900);
```

**Review by AaronVanGeffen — COMMENTED** [review]
Thanks! Looks good to me, save for the BOM insertions.

File: src/OpenLoco/Title.cpp (hunk 2 above)
Comment (ZehMatt) [reply src/OpenLoco/Title.cpp]: Yeah sorry about that, #777 will prevent this in the future.

File: src/OpenLoco/Scenario.cpp [hunk 5]
```diff
@@ -24,10 +31,93 @@ namespace OpenLoco::Scenario
...
+    }
+
+    // 0x0043C8FD
+    static void sub_43C8FD()
+    {
+        call(0x0043C8FD);
+    }
+
     // 0x0043C88C
     void reset()
```
[the remark this answers is not in the document]
Comment (ZehMatt) [reply src/OpenLoco/Scenario.cpp:93]: Let me check in IDA again, I thought all references are implemented, but you are probably right.

[the remark this answers is not in the document]
Comment (ZehMatt) [reply src/OpenLoco/Scenario.cpp:93]: You are right as expected, I'll add the hook.

**Review by AaronVanGeffen — APPROVED** [review]

**Pull request merged**